src/libsystemd/sd-device/device-enumerator.c

```diff
@@ -24,6 +24,16 @@ typedef enum DeviceEnumerationType {
         _DEVICE_ENUMERATION_TYPE_INVALID = -EINVAL,
 } DeviceEnumerationType;
 
+typedef enum MatchFlag {
+        MATCH_NONE        = 0,
```

Comment:
`test_matches()` checks several conditions even if the flag is zero.
But, in this case, we'd like to add all parents without checking any conditions.
So, maybe better to do something like the following:
```c
typedef enum MatchFlag {
        MATCH_NONE        = 0,
        MATCH_BASIC       = 1u << 0,
        MATCH_SYSNAME     = 1u << 1,
        MATCH_SUBSYSTEM   = 1u << 2,
        MATCH_PARENT      = 1u << 3,
        MATCH_TAG         = 1u << 4,

        MATCH_ALL         = (1u << 5) - 1,
} MatchFlag;
```
Then, make the unconditional checks in `test_matches()` should be only done when `MATCH_BASIC` is set.

Reply:
I've checked MatchFlag use is almost always used like (MATCH_ALL & (~ something)) so this should work just fine, and it looks good to me.

But, can you just recheck the test_matches() to see if I have correctly captured your intention?